src/xmipp/libraries/reconstruction/angular_distance.cpp

```diff
@@ -573,3 +625,114 @@ void ProgAngularDistance::computeWeights()
     DF2weighted.removeDisabled();
     DF2weighted.write(fn_out+"_weights.xmd");
 }
+
+void ProgAngularDistance::euler2quat(   double rot, double tilt, double psi,
+                                        double q[4] )
+{
+    // Convert to radians and divide by 2
+    const double ai = -DEG2RAD(rot) / 2;
+    const double aj = DEG2RAD(tilt) / 2;
+    const double ak = -DEG2RAD(psi) / 2;
+
+    // Obtain sin and cos
+    const double    si = std::sin(ai),
+                    ci = std::cos(ai),
+                    sj = std::sin(aj),
+                    cj = std::cos(aj),
+                    sk = std::sin(ak),
+                    ck = std::cos(ak);
+
+    const double    cc = ci * ck,
+                    cs = ci * sk,
+                    sc = si * ck,
+                    ss = si * sk;
+
+    // Compute the quaternion values
+    // WXYZ
+    q[0] = +cj * (cc - ss);
+    q[1] = +sj * (cs - sc);
+    q[2] = -sj * (cc + ss);
+    q[3] = +cj * (cs + sc);
+}
+
+void ProgAngularDistance::quat2Euler(   const double q[4],
+                                        double& rot, double& tilt, double& psi )
+{
+    // Multiply the quaternion by sqrt2 to avoid multiplying pairs by 2
+    const auto sqrt2 = std::sqrt(2);
+    const auto qw = sqrt2 * q[0];
+    const auto qx = sqrt2 * q[1];
+    const auto qy = sqrt2 * q[2];
+    const auto qz = sqrt2 * q[3];
+
+    // Obtain pairwise products of the quaternion elements
+    const auto qx2 = qx*qx;
+    const auto qy2 = qy*qy;
+    const auto qz2 = qz*qz;
+    const auto qxqy = qx*qy;
+    const auto qxqz = qx*qz;
+    const auto qxqw = qx*qw;
+    const auto qyqz = qy*qz;
+    const auto qyqw = qy*qw;
+    const auto qzqw = qz*qw;
+
+    // Ensemble the matrix
+    const double    m00 = 1 - qy2 - qz2,
```

Comment:
m00 and m01 are never read

Reply:
I'm going to re-structure the code so that only necessary elements  are computed. In any case, any modern compiler should be able to do it.

Reply:
Solved in 7f6c5f5